src/fides/api/ops/models/privacy_preference.py

```diff
@@ -110,6 +115,29 @@ class PrivacyPreferenceHistory(Base):
         uselist=False,
     )
 
+    def cache_system_status(self, system: str, status: ExecutionLogStatus) -> None:
+        """Update the cached affected system status for consent reporting
+
+        Typically this should just be called for consent connectors only.
+        If no request is made or email is sent, this should be called with a status of skipped.
+        """
+        db = Session.object_session(self)
```

Comment:
What's the reason this is using `Session.object_session` vs. the conventional `get_db`?

Reply:
@seanpreston thanks for flagging, you're right it stands out. I'm calling this in the saas connector where we don't have a db session available at the moment.  Traditionally the graphs did all the querying up front and it was just the execution logs that were often writing to the db as we went.   

On closer inspection, it does look like `TaskResources` possesses the Session, so I could pass this into the `run_consent_request` call instead and then pass that along to this method in turn.